torch/_dynamo/side_effects.py

```diff
@@ -246,8 +246,9 @@ def track_global_existing(self, source: Source, item: Any):
         self.keepalive.append(item)
         return variable
 
-    def prune_dead_object_new(self, tx):
-        live_new_objects = set()
+    def prune_dead_object_new(self, tx, live_new_objects=None):
+        if live_new_objects is None:
+            live_new_objects = set()
```

Comment:
Careful, do you want to mutate the input?

Reply:
Why not, isn't this a harmless pattern for setting the default of what is now an optional param `live_new_objects`? You either pass a set explicitly, or you omit it and it gets initialized as before.

Reply:
Oh I see where this question is coming from now, and I think it's related to the @jansel question below as well. I *do* want to mutate this set. This is just an initial set, it is expected to grow. The intent is that *at least* the objects is this set will not be pruned, although fewer objects might end up being pruned because of additional marking in this function as before.